### DISCUSSION TITLE
Rearrangement and insertion of codes

### DISCUSSION THREAD

**Discussion by P-Sian** [ccbogel/QualCoder#910] [Q&A] [answered]
Hello, I have just begun using QualCoder for coding. I have 2 questions:

1) After I set up a bunch of codes, I noticed that they only follow the order alphabetically or numerically. Say, I want my codes to be in this particular order (AYCUB), it seems the only way I can achieve such is to assign prefixes to them, like 1.A, 2. Y, 3. C, 4.U, 5.B, or a.A, b.Y, c.C, d.U, e.B. Otherwise, they will automatically become ABCUY.

May be there is a way to achieve AYCUB without the prefixes - if so, can anyone chime in? If this issue has already been discussed and resolved, apologies, please point me to the discussion so I can check out the dialogues there.

2) My another question is about inserting/adding a new code into the preexisting set of codes in the order I want (i.e. AYCUB), let say I want to insert an "E" to make the set of codes to be like AYECUB. In reality, since the codes automatically arrange themselves into ABCUY, meaning, after adding the new code "E", the set will become ABCEUY, not AYECUB.

The only way I know to solve the problem is again by adding prefixes so they will be in the order of 1.A, 2.Y, 3.E, 4.C, 5.U, 6.B. Is there another way without using the prefixes? But the prefix is a tedious method, it means, I need to rename all the codes after the new code I inserted in. With a large amount of codes and a large data set, this is a nightmare.

Thank you.

**Comment by ccbogel** [comment, marked as answer]
At this stage I think prefixes are the only way. 
They are sorted alphabetically, not numerically, so: 1, 11, 2, 20. 
I would suggest for numbers up to 9 doing 01 02 03. So sorting would be like this: 01, 02, 11, 20. 
But if you have hundreds of codes a 00 prefix would be better: so 001, 002, 010, 020, etc
I would also suggest adding a 0 at the end of each number, so: 010, 020, 110, 200. This to prepare for adding more codes in between later on.
Then you could slip in some codes between 010 and 020: 015, 013
